pkg/apis/aerogear/v1alpha1/types.go

```diff
@@ -43,80 +43,81 @@ type KeycloakSpec struct {
-	ServiceAccountsEnabled    bool                     `json:"serviceAccountsEnabled"`
-	PublicClient              bool                     `json:"publicClient"`
-	FrontchannelLogout        bool                     `json:"frontchannelLogout"`
+	SurrogateAuthRequired     bool                     `json:"surrogateAuthRequired,omitempty"`
+	Enabled                   bool                     `json:"enabled,omitempty"`
+	ClientAuthenticatorType   string                   `json:"clientAuthenticatorType,omitempty"`
+	DefaultRoles              []string                 `json:"defaultRoles,omitempty,omitempty"`
+	RedirectUris              []string                 `json:"redirectUris,omitempty"`
+	WebOrigins                []string                 `json:"webOrigins,omitempty"`
+	NotBefore                 int                      `json:"notBefore,omitempty"`
+	BearerOnly                bool                     `json:"bearerOnly,omitempty"`
+	ConsentRequired           bool                     `json:"consentRequired,omitempty"`
+	StandardFlowEnabled       bool                     `json:"standardFlowEnabled,omitempty"`
+	ImplicitFlowEnabled       bool                     `json:"implicitFlowEnabled,omitempty"`
+	DirectAccessGrantsEnabled bool                     `json:"directAccessGrantsEnabled,omitempty"`
+	ServiceAccountsEnabled    bool                     `json:"serviceAccountsEnabled,omitempty"`
+	PublicClient              bool                     `json:"publicClient,omitempty"`
+	FrontchannelLogout        bool                     `json:"frontchannelLogout,omitempty"`
 	Protocol                  string                   `json:"protocol,omitempty"`
-	Attributes                map[string]string        `json:"attributes"`
-	FullScopeAllowed          bool                     `json:"fullScopeAllowed"`
-	NodeReRegistrationTimeout int                      `json:"nodeReRegistrationTimeout"`
-	ProtocolMappers           []KeycloakProtocolMapper `json:"protocolMappers"`
-	UseTemplateConfig         bool                     `json:"useTemplateConfig"`
-	UseTemplateScope          bool                     `json:"useTemplateScope"`
-	UseTemplateMappers        bool                     `json:"useTemplateMappers"`
-	Access                    map[string]string        `json:"access"`
+	Attributes                map[string]string        `json:"attributes,omitempty"`
+	FullScopeAllowed          bool                     `json:"fullScopeAllowed,omitempty"`
+	NodeReRegistrationTimeout int                      `json:"nodeReRegistrationTimeout,omitempty"`
+	ProtocolMappers           []KeycloakProtocolMapper `json:"protocolMappers,omitempty"`
+	UseTemplateConfig         bool                     `json:"useTemplateConfig,omitempty"`
+	UseTemplateScope          bool                     `json:"useTemplateScope,omitempty"`
+	UseTemplateMappers        bool                     `json:"useTemplateMappers,omitempty"`
+	Access                    map[string]string        `json:"access,omitempty"`
```

Comment:
@dimitraz @vchepeli fyi, this resolves the Conflict issue when creating more than 1 keycloak resource (user/client etc..) . We just need to make sure we aren't posting up empty values during the create.

You can remove that UUID generation thing you were doing since the server does create that correctly as you would expect.

Reply:
:+1: